engine/lib/ast.ml

```diff
@@ -370,11 +372,13 @@ functor
         }
 
     and loop_state = { init : expr; bpat : pat; witness : F.state_passing_loop }
+
     (* | WhileLoop of { *)
     (*     condition: expr; *)
     (*     witness : F.while_loop; *)
     (*   } *)
 
+    (* TODO: LHS should be places or "compositions" of places, see [assignee expression] in https://doc.rust-lang.org/reference/expressions.html#place-expressions-and-value-expressions  *)
```

Comment:
Is there an issue for this? If not, please file one. And reference here in any case.